### ISSUE TITLE
apitrace doesn't work with go binaries and go-gl wrappers

### ISSUE THREAD

**Issue by dlespiau** [apitrace/apitrace#431]
Test binary: glfw31-gl21-cube in https://github.com/go-gl/examples
This binary is using the go-gl wrappers (https://github.com/go-gl/gl).

I've tried:

  - apitrace trace
  - LD_PRELOAD with the glx .so
  - The ln -s libGL.so.1 method + LD_LIBRARY_PATH and TRACE_LIBGL

I can actually see api trace outputting some logs:

    $ apitrace trace ./glfw31-gl21-cube 
    apitrace: loaded into /usr/bin/apitrace
    apitrace: unloaded from /usr/bin/apitrace
    apitrace: loaded into /home/damien/go/src/github.com/go-gl/examples/glfw31-gl21-cube/glfw31-gl21-cube
    apitrace: tracing to /home/damien/go/src/github.com/go-gl/examples/glfw31-gl21-cube/glfw31-gl21-cube.trace
    apitrace: attempting to read configuration file: /home/damien/.config/apitrace/gltrace.conf
    apitrace: warning: unknown function "glAccumxOES"
    apitrace: warning: unknown function "glAlphaFuncxOES"
    apitrace: warning: unknown function "glBitmapxOES"
    ...

but the .trace file is 0 bytes.

**Comment by jrfonseca** [comment]
The trace will grow if you wait long enough.

Note that apitrace buffers data when tracing, and flushes to disk when the app exits, or there's a signal.

The problem here is that:
1. the demo appears to have no clean way of exiting: Ctrl-C or closing the window all appear to cause the app to terminate abrutely (without calling any destructors)
2. go runtime or something else is masking the signals

Net result is that 

I'm afraid there's nothing that can be done from apitrace side:
- we need to buffer data in order to compress it
- things like clean termination or overriding signal handlers is completely withing the application control, which apitrace can't change.  apitrace only sets the signal handlers once at startup, it cant be overriding signal handlers all the time as that could interfere with application

In conclusion, apitrace **does** work with go binaries and go-gl.  One just needs to run them long enough, or fix them so they terminate cleanly.

**Comment by dlespiau** [comment]
Ah yes, waiting a bit longer did work. Sorry for the noise.
